crates/burn-dataset/src/vision/image_folder.rs

```diff
@@ -129,15 +130,77 @@ struct PathToImageDatasetItem {
     classes: HashMap<String, usize>,
 }
 
+fn image_path_to_vec_pixel_depth(image_path: &PathBuf) -> Vec<PixelDepth> {
+    // Load image from disk
+    let image = image::open(image_path).unwrap();
+
+    // Image as Vec<PixelDepth>
+    let img_vec = match image.color() {
+        ColorType::L8 => image
+            .into_luma8()
+            .iter()
+            .map(|&x| PixelDepth::U8(x))
+            .collect(),
+        ColorType::La8 => image
+            .into_luma_alpha8()
+            .iter()
+            .map(|&x| PixelDepth::U8(x))
+            .collect(),
+        ColorType::L16 => image
+            .into_luma16()
+            .iter()
+            .map(|&x| PixelDepth::U16(x))
+            .collect(),
+        ColorType::La16 => image
+            .into_luma_alpha16()
+            .iter()
+            .map(|&x| PixelDepth::U16(x))
+            .collect(),
+        ColorType::Rgb8 => image
+            .into_rgb8()
+            .iter()
+            .map(|&x| PixelDepth::U8(x))
+            .collect(),
+        ColorType::Rgba8 => image
+            .into_rgba8()
+            .iter()
+            .map(|&x| PixelDepth::U8(x))
+            .collect(),
+        ColorType::Rgb16 => image
+            .into_rgb16()
+            .iter()
+            .map(|&x| PixelDepth::U16(x))
+            .collect(),
+        ColorType::Rgba16 => image
+            .into_rgba16()
+            .iter()
+            .map(|&x| PixelDepth::U16(x))
+            .collect(),
+        ColorType::Rgb32F => image
+            .into_rgb32f()
+            .iter()
+            .map(|&x| PixelDepth::F32(x))
+            .collect(),
+        ColorType::Rgba32F => image
+            .into_rgba32f()
+            .iter()
+            .map(|&x| PixelDepth::F32(x))
+            .collect(),
+        _ => panic!("Unrecognized image color type"),
```

Comment:
Following the previous comment, I don't think we should support floating point segmentation masks. Images with alpha channels should probably not be supported as mask input either, so no `La8`, `La16`, `Rbga8`, `Rgba16` on top of that 🤔 

With these changes the function can probably be renamed since it will be specifically parsing segmentation mask input images to return the vec of class identifiers (i.e., indices). Probably want to make sure the returned vec is represents a single channel too since a segmentation mask is just 2D with each (x,y) pixel representing a class.

Reply:
modified function name to `segmentation_mask_to_vec_usize` and only support luma8, luma16, rgb8, and rgb16. fixed in next commit.